src/crepe/api/LoopManager.cpp

```diff
@@ -0,0 +1,60 @@
+
+#include "../system/RenderSystem.h"
+#include "../system/ScriptSystem.h"
+#include "../facade/SDLContext.h"
+
+#include "LoopManager.h"
+#include "LoopTimer.h"
+
+namespace crepe {
+
+void LoopManager::process_input() {
+    SDLContext::get_instance().handle_events(this->game_running);
+}
+void LoopManager::start(){
+    this->setup();
+    this->loop();
+}
+void LoopManager::set_running(bool running) { this->game_running = running; }
+
+void LoopManager::fixed_update() {
+}
+
+void LoopManager::loop() {
+    LoopTimer & timer = LoopTimer::get_instance();
+    timer.start();
+
+    while (game_running) {
+        timer.update();
+
+        while (timer.get_lag() >= timer.get_fixed_delta_time()) {
+            process_input();
```

Comment:
`process_input`, `fixed_update`, `update`, en `render` moeten allemaal met `this->` aangeroepen worden omdat het members zijn